### PR TITLE
212 create model renderer api

### LINKED ISSUE #26: Create ModelRenderer API
To make it possible to manipulate body parts of single entities easily, it is necessary to write an API for the "ModelRenderer" of Minecraft

### PR DESCRIPTION
**Pull request by jan-br**
Closes https://github.com/FlintMC/FlintMC/issues/26

### CODE REVIEW

File: mcapi/src/internal/java/net/flintmc/mcapi/internal/render/DefaultMinecraftRenderMeta.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package net.flintmc.mcapi.internal.render;
+
+import net.flintmc.framework.inject.assisted.AssistedInject;
+import net.flintmc.framework.inject.implement.Implement;
+import net.flintmc.mcapi.render.MinecraftRenderMeta;
+import net.flintmc.util.math.matrix.Matrix3x3f;
+import net.flintmc.util.math.matrix.Matrix4x4f;
+
+@Implement(MinecraftRenderMeta.class)
+public class DefaultMinecraftRenderMeta implements MinecraftRenderMeta {
```
Comment (R0bbyYT) [inline mcapi/src/internal/java/net/flintmc/mcapi/internal/render/DefaultMinecraftRenderMeta.java]: The `@Override` annotations and `{@inheritDoc}`'s are missing

File: mcapi/src/main/java/net/flintmc/mcapi/entity/render/EntityModelBoxHolder.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package net.flintmc.mcapi.entity.render;
+
+import net.flintmc.framework.inject.assisted.Assisted;
+import net.flintmc.framework.inject.assisted.AssistedFactory;
+import net.flintmc.mcapi.entity.Entity;
+import net.flintmc.render.model.ModelBoxHolder;
+
+public interface EntityModelBoxHolder extends ModelBoxHolder<Entity, EntityRenderContext> {
```
Comment (R0bbyYT) [inline mcapi/src/main/java/net/flintmc/mcapi/entity/render/EntityModelBoxHolder.java:31]: Documentation is missing

File: mcapi/src/main/java/net/flintmc/mcapi/entity/render/EntityRenderContext.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+import net.flintmc.framework.inject.assisted.Assisted;
+import net.flintmc.framework.inject.assisted.AssistedFactory;
+import net.flintmc.mcapi.entity.Entity;
+import net.flintmc.mcapi.render.MinecraftRenderContext;
+import net.flintmc.render.model.ModelBoxHolder;
+
+import java.util.Map;
+
+public interface EntityRenderContext
```
Comment (R0bbyYT) [inline mcapi/src/main/java/net/flintmc/mcapi/entity/render/EntityRenderContext.java:33]: Documentation is missing

File: mcapi/src/main/java/net/flintmc/mcapi/render/MinecraftRenderMeta.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package net.flintmc.mcapi.render;
+
+import net.flintmc.framework.inject.assisted.AssistedFactory;
+import net.flintmc.util.math.matrix.Matrix3x3f;
+import net.flintmc.util.math.matrix.Matrix4x4f;
+
+public interface MinecraftRenderMeta {
```
Comment (R0bbyYT) [inline mcapi/src/main/java/net/flintmc/mcapi/render/MinecraftRenderMeta.java:32]: Documentation is missing

File: mcapi/src/v1_15_2/java/net/flintmc/mcapi/v1_15_2/entity/render/ModelRendererInterceptorOld.java [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+import net.flintmc.mcapi.render.MinecraftRenderMeta;
+import net.flintmc.render.model.ModelBoxHolder;
+import net.flintmc.transform.javassist.ClassTransform;
+import net.flintmc.transform.javassist.ClassTransformContext;
+import net.flintmc.transform.javassist.CtClassFilter;
+import net.flintmc.transform.javassist.CtClassFilters;
+import net.flintmc.util.mappings.ClassMappingProvider;
+
+@Singleton
+public class ModelRendererInterceptorOld {
```
Comment (R0bbyYT) [inline mcapi/src/v1_15_2/java/net/flintmc/mcapi/v1_15_2/entity/render/ModelRendererInterceptorOld.java]: I think this class should not be committed with?

File: render/model-renderer/src/internal/java/net/flintmc/render/model/internal/DefaultModelBox.java [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+import org.joml.Vector3f;
+
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+
+@Implement(ModelBox.class)
+public class DefaultModelBox implements ModelBox {
```
Comment (R0bbyYT) [inline render/model-renderer/src/internal/java/net/flintmc/render/model/internal/DefaultModelBox.java:32]: The `@Override` annotations and `{@inheritDoc}`'s are missing

File: render/model-renderer/src/internal/java/net/flintmc/render/model/internal/DefaultRendererRepository.java [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+import net.flintmc.framework.inject.primitive.InjectionHolder;
+import net.flintmc.render.model.*;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Predicate;
+
+@Singleton
+@Implement(RendererRepository.class)
+public class DefaultRendererRepository implements RendererRepository {
```
Comment (R0bbyYT) [inline render/model-renderer/src/internal/java/net/flintmc/render/model/internal/DefaultRendererRepository.java:33]: The `@Override` annotations and `{@inheritDoc}`'s are missing

File: render/model-renderer/src/main/java/net/flintmc/render/model/ModelBox.java [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+import net.flintmc.framework.inject.assisted.Assisted;
+import net.flintmc.framework.inject.assisted.AssistedFactory;
+import org.joml.Vector2f;
+import org.joml.Vector3f;
+
+import java.awt.*;
+import java.util.Collection;
+import java.util.List;
+
+public interface ModelBox {
```
Comment (R0bbyYT) [inline render/model-renderer/src/main/java/net/flintmc/render/model/ModelBox.java:27]: Documentatino is missing.

File: util/math/src/main/java/net/flintmc/util/math/matrix/Matrix3x3.java [hunk 9]
```diff
@@ -0,0 +1,246 @@
+package net.flintmc.util.math.matrix;
+
```
Comment (R0bbyYT) [inline util/math/src/main/java/net/flintmc/util/math/matrix/Matrix3x3.java:21]: Documentation is missing

File: mcapi/src/main/java/net/flintmc/mcapi/render/MinecraftRenderContext.java [hunk 10]
```diff
@@ -0,0 +1,17 @@
+package net.flintmc.mcapi.render;
+
+import net.flintmc.render.model.RenderContext;
+import net.flintmc.render.model.RenderContextAware;
+import net.flintmc.render.model.Renderable;
+
```
Comment (R0bbyYT) [inline mcapi/src/main/java/net/flintmc/mcapi/render/MinecraftRenderContext.java:25]: Documentation is missing.

**Comment by Janrupf** [comment]
Please re-visit some of the commits and their respective messages and rename/squash them where appropriate.

**Review by Janrupf — CHANGES_REQUESTED** [review]
Please re-visit some of the commits and their respective messages and rename/squash them where appropriate.

File: mcapi/src/internal/java/net/flintmc/mcapi/internal/render/DefaultMinecraftRenderMeta.java (hunk 1 above)
Comment (jan-br) [reply mcapi/src/internal/java/net/flintmc/mcapi/internal/render/DefaultMinecraftRenderMeta.java]: renamed to VersionedMinecraftRenderMeta.

File: mcapi/src/v1_15_2/java/net/flintmc/mcapi/v1_15_2/entity/render/ModelRendererInterceptorOld.java (hunk 5 above)
Comment (jan-br) [reply mcapi/src/v1_15_2/java/net/flintmc/mcapi/v1_15_2/entity/render/ModelRendererInterceptorOld.java]: already deleted

**Comment by jan-br** [comment]
> Please re-visit some of the commits and their respective messages and rename/squash them where appropriate.

Simply cant get git squash to work there. "Commit has 2 parents". Please take a look at it @Janrupf

**Review by Janrupf — APPROVED** [review]
Decided to squash merge

**Pull request merged**